src/commands/remote_signer_setup.py

```diff
@@ -126,28 +137,72 @@ async def main(output_dir: Path) -> None:
             Web3.to_hex(bls.SkToPk(priv_key)) for priv_key in private_key_shares
         ]
 
-    _export_keystores(
-        credentials=credentials,
-        keystores_dir=output_dir,
-        password_file=str(settings.keystores_password_file),
-        per_keystore_password=False,
-    )
-
     click.echo(
         f'Successfully generated {len(credentials)} key shares'
         f' for {len(keystores)} private key(s)!',
     )
 
-    # Remove local keystores - remote signer will be used
-    for keystore_file in os.listdir(settings.keystores_dir):
-        if not keystore_file.startswith('keystore-'):
-            continue
-        os.remove(settings.keystores_dir / keystore_file)
+    # Import as keystores to remote signer
+    key_share_keystores = []
+    for credential in credentials:
+        key_share_keystores.append(
+            deepcopy(
```

Comment:
It's because of this class attribute in the staking-deposit-cli codebase:

https://github.com/ethereum/staking-deposit-cli/blob/33cdafe64f58bd885ceac0fe05edbaa0359e9241/staking_deposit/key_handling/keystore.py#L84

The `crypto` class attribute was being shared across all `credential` instances, being overwritten during the for loop and all keystores ended up having the same value (of the last keystore of the for loop). The deepcopy is there to preserve the value of the `crypto` attribute for each keystore.

Tests caught this, I never would have expected this and it was fun trying to find why only one pubkey was being added to the remote signer.

I've added an explicit test for this behavior here: https://github.com/stakewise/v3-operator/pull/146/commits/f2d6e7b11669f5fc4b99f7427a7be2454af1116d

However testing for this did require a bit of an ugly try-except statement. I don't see a much cleaner way out of that aside from switching to AsyncClick (not ideal and probably too big a change) or not explicitly testing for this behavior (may be acceptable to you, it is unlikely anything similar will pop up but it may cause a regression in the future).